### PR TITLE
Updated entity types and implemented post services

### PR DESCRIPTION
**Pull request by lucass0717**
- Refactored recipe and post types to align with node server schema
- Implemented create post back end logic
- Render fetched posts JSON on the feed page

### CODE REVIEW

File: src/data/posts/postsData.ts [hunk 1]
```diff
@@ -1,31 +1,33 @@
 import PostType, {FoodGroup} from "../../modules/postType";
-
 const posts : Array<PostType> = [
   {
     "_id": 10,
-    "recipe_id": 715467,
-    "user_id": 1,
-    "caption": "10/10 would make again!",
+    "recipeId": "643b7f955da0d4402e90938e",
+    "userId": "1",
```
Comment (surbhi-gulati) [inline src/data/posts/postsData.ts:6]: should we add more variance to these posts than the userId? or is the purpose of this to repost the same recipe to the same group multiple times and illustrate the post:recipe ratio can be something other than 1:1?

File: src/modules/userType.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 export default interface UserType {
-  id: number,
+  _id: string,
```
Comment (surbhi-gulati) [inline src/modules/userType.ts:2]: reason?

File: src/services/post-services.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  console.log(post);
+  console.log(recipe);
+
+  try {
+    const response = await api.post(`${POSTS_URL}`, {
+      post,
+      recipe
+    });
+    post = response.data;
```
Comment (surbhi-gulati) [inline src/services/post-services.ts:17]: nit: combine line 17 and 18 to `return response.data`?

File: src/components/posts/post-card.tsx [hunk 4]
```diff
@@ -16,8 +16,8 @@ export const PostCard = (props: PostType) => {
     dispatch(updatePostLikes(id))
   }
 
-  let user = sampleUsers.find(user => user.id === props.user_id);
-  let recipe = recipesArray.find(recipe => recipe.id === props.recipe_id);
+  let user = sampleUsers.find(user => user._id === props.userId);
```
Comment (surbhi-gulati) [inline src/components/posts/post-card.tsx:19]: If validation is completed we should revert to `recipe.id = props.recipeId` instead of hardcoding 8652

**Review by surbhi-gulati — CHANGES_REQUESTED** [review]

File: src/components/posts/post-card.tsx (hunk 4 above)
Comment (bellehlim) [reply src/components/posts/post-card.tsx:19]: this is gonna be replaced in the future w services, just for sake of debugging

File: src/data/posts/postsData.ts (hunk 1 above)
Comment (lucass0717) [reply src/data/posts/postsData.ts:6]: These were just used to test post rendering, they will be deprecated eventually.

File: src/modules/userType.ts (hunk 2 above)
Comment (lucass0717) [reply src/modules/userType.ts:2]: Mongo DB ObjectId type is a 24 character string

File: src/services/post-services.ts (hunk 3 above)
Comment (lucass0717) [reply src/services/post-services.ts:17]: was for readability to show what the response is, but that can just be documented with a comment

File: src/components/posts/post-card.tsx (hunk 4 above)
Comment (lucass0717) [reply src/components/posts/post-card.tsx:19]: yeah this was a dirty fix to test rendering as well

File: src/modules/userType.ts (hunk 2 above)
Comment (surbhi-gulati) [reply src/modules/userType.ts:2]: gotcha, thanks!

**Review by surbhi-gulati — APPROVED** [review]

**Pull request merged**